check for keys containing password

whatsadebugger opened the pull request:
small follow up to the last blacklist pr

dimroc commented on web/router.go:
```diff
@@ -315,7 +315,7 @@ func readSanitizedJSON(buf *bytes.Buffer) (string, error) {
 	cleaned := map[string]interface{}{}
 	for k, v := range dst {
-		if _, ok := blacklist[strings.ToLower(k)]; ok {
+		if _, ok := blacklist[strings.ToLower(k)]; ok || strings.Contains(k, "password") {
```
Let's also `ToLower` it before checking `Contains`:
```
lowerk := strings.ToLower(k)
...
```

dimroc commented on web/router.go:
```diff
@@ -315,7 +315,8 @@ func readSanitizedJSON(buf *bytes.Buffer) (string, error) {
 	for k, v := range dst {
-		if _, ok := blacklist[strings.ToLower(k)]; ok {
+		lowerk := strings.ToLower(k)
+		if _, ok := blacklist[lowerk]; ok || strings.Contains(lowerk, "password") {
```
Can we also add this to function directly below, `redact`?

dimroc replied:
If there's an opportunity to DRY this up, please do. There may be some challenges around coercing types.

whatsadebugger commented:
I was thinking of doing something like this to dry up those for loops but I need some feedback.
```
func clean(values map[string]interface{}) map[string]interface{} {
       cleaned := map[string]interface{}{}
        for k, v := range values {
               if isBlacklisted(k) {
                       switch v.(type) {
                       case []string:
                               cleaned[k] = []string{"*REDACTED*"}
                       case string:
                               cleaned[k] = "*REDACTED*"
                       }
                        continue
                }
                cleaned[k] = v
        }
       return cleaned.Encode()
}
```

whatsadebugger commented:
So the latest commit works but the conversions for the url redaction are ugly. I'm not sure if this can be any easier.

```
2019-01-23T13:58:53Z [INFO]  GET /bridges                                       web/router.go:255       clientIP=::1 latency=278.276µs method=GET path=/bridges query=12=2&current_password=%2AREDACTED%2A&new_account_password=%2AREDACTED%2A&password=%2AREDACTED%2A&password1=%2AREDACTED%2A&wow=wow servedAt=2019-01-23 07:58:53 status=200 
```

whatsadebugger commented:
This pr should be ready @dimroc
